crates/matrix-sdk/src/event_cache/room/events.rs

```diff
@@ -44,12 +51,22 @@ impl Default for RoomEvents {
 impl RoomEvents {
     /// Build a new [`RoomEvents`] struct with zero events.
     pub fn new() -> Self {
-        Self { chunks: LinkedChunk::new(), deduplicator: Deduplicator::new() }
+        let mut chunks = LinkedChunk::new_with_update_history();
+        let chunks_updates_as_vectordiffs = chunks
+            .as_vector()
+            // SAFETY: The `LinkedChunk` has been built with `new_with_update_history`, so
+            // `as_vector` must return `Some(…)`.
+            .expect("`LinkedChunk` must have been constructor with `new_with_update_history`");
```

Comment:
To avoid this expect, would it make sense that `new_with_update_history()` return the AsVector data structure too?

Reply:
Hmm, `new_with_update_history()` could return `(LinkedChunk, Update)`, but I don't see why it should return an `AsVector` directly. Problem is: to compute an `AsVector`, it needs an `&mut Update`, so the constructor would have to return `(LinkedChunk, &mut Update)`, where the `&mut` is a mutable reference to `LinkedChunk` itself. Er, that's not possible.

Reply:
Ok, too bad!